1-mandatory/3-slideshow/slideshow.js

```diff
@@ -1 +1,54 @@
 // Write your code here
+var count = 0;
+var backWardInterval;
+var forWardInterval;
+var selectedTimeForEachSlide;
+function forward() {
+  count = (count + 1) % 10;
+  document.getElementById("image").src = "assets/images/pic" + count + ".jpg";
+}
+
+function backward() {
+  count = (count - 1) % 10;
+  if (count === -1) count = 9;
+  document.getElementById("image").src = "assets/images/pic" + count + ".jpg";
+}
+
+function autoForward() {
+  clearInterval(backWardInterval);
+  forWardInterval = setInterval(function () {
+    console.log(selectedTimeForEachSlide);
+    count = (count + 1) % 10;
+    document.getElementById("image").src = "assets/images/pic" + count + ".jpg";
+  }, selectedTimeForEachSlide);
+}
+
+function autoBackward() {
+  clearInterval(forWardInterval);
+  backWardInterval = setInterval(function () {
+    console.log(selectedTimeForEachSlide);
+    count = (count - 1) % 10;
+    if (count === -1) count = 9;
+    document.getElementById("image").src = "assets/images/pic" + count + ".jpg";
```

Comment:
There's a lot of repeated/duplicate logic in these 4 functions. Can you extract some of it out into more functions?

Reply:
Yes sure why not,